mithril-aggregator/src/artifact_builder/cardano_stake_distribution.rs

```diff
@@ -0,0 +1,100 @@
+use anyhow::anyhow;
+use async_trait::async_trait;
+use std::sync::Arc;
+
+use mithril_common::{
+    entities::{CardanoStakeDistribution, Certificate, Epoch},
+    StdResult,
+};
+use mithril_persistence::store::StakeStorer;
+
+use crate::ArtifactBuilder;
+
+/// A [CardanoStakeDistributionArtifact] builder
+pub struct CardanoStakeDistributionArtifactBuilder {
+    stake_store: Arc<dyn StakeStorer>,
```

Comment:
Maybe you should use a `StakeDistributionRetriever` instead of a `StakeStoer` trait in the artifact builder?

Reply:
Done ✅